physics/ephemeris_body.hpp

```diff
@@ -1399,7 +1401,8 @@ Ephemeris<Frame>::ComputeGravitationalAccelerationBetweenAllMassiveBodies(
     Instant const& t,
     std::vector<Position<Frame>> const& positions,
     std::vector<Vector<Acceleration, Frame>>& accelerations) const {
-  RETURN_IF_STOPPED;
+  // Do not RETURN_IF_STOPPED here, it's too hard to undo the state changes made
+  // half way through the loop of the integrator.
```

Comment:
That comment should be made in https://github.com/mockingbirdnest/Principia/blob/master/integrators/ordinary_differential_equations.hpp too, and the UpdateWithAbort should be changed to Update.